### PR TITLE
Add a condition so we can target specific user ids or usernames

### PR DESCRIPTION
**Pull request by golenkovm**
This is to implement #38

### CODE REVIEW

**Comment by golenkovm** [comment]
#40 needs to be merged first

File: classes/form/edit_conditions.php [hunk 1]
```diff
@@ -153,6 +162,19 @@ public function definition() {
             )
         );
 
+        $repeatarray[] = $mform->createElement(
```
Comment (Peterburnett) [inline classes/form/edit_conditions.php:172]: Could we swap this to the user search w/ Ajax selector? will be much easier to use

Comment (golenkovm) [reply classes/form/edit_conditions.php:172]: Good point, it looks much better now.

Comment (golenkovm) [reply classes/form/edit_conditions.php:172]: Hang on, I'm getting `Exception - Serialization of 'Closure' is not allowed` in some scenario. Will have a look what's happening there

Comment (golenkovm) [reply classes/form/edit_conditions.php:172]: Sadly this was revealed when everything else is implemented :(
It seems like moodle form doesn't like cloning elements with functions https://github.com/moodle/moodle/blob/master/lib/moodlelib.php#L9430

```
Stack trace:
line 9430 of /lib/moodlelib.php: Exception thrown
line 9430 of /lib/moodlelib.php: call to serialize()
line 1178 of /lib/formslib.php: call to fullclone()
line 213 of /admin/tool/abconfig/classes/form/edit_conditions.php: call to moodleform->repeat_elements()
line 214 of /lib/formslib.php: call to tool_abconfig\form\edit_conditions->definition()
line 50 of /admin/tool/abconfig/edit_conditions.php: call to moodleform->__construct()
```

**Comment by golenkovm** [comment]
Some screenshots from my dev box:

image

image

image

File: classes/form/edit_conditions.php (hunk 1 above)
Comment (golenkovm) [reply classes/form/edit_conditions.php:172]: Ok, this seems to be sorted by moving the callable function into a separate static method.
Happy to have this reviewed @Peterburnett

**Pull request merged**